notebook_analyzer/connector/get_data.py

```diff
@@ -45,11 +46,30 @@ def metadata(self):
     def cells(self):
         return self._cells
 
+    def download_notebook(self):
+        try:
+            notebook_string = self.download_notebook_amazon()
+
+        except urllib.error.URLError:
+            notebook_string = self.download_notebook_file()
+
+        return notebook_string
```

Comment:
This is optional.
ulrlib is quite old tech - we could use somthing more updated. We are dooing it 10m times, so we could we here a little time.  Look here for benchmark https://github.com/svanoort/python-client-benchmarks 

If you deicde to update - please do it in separete PR